sbpy/photometry/core.py

```diff
@@ -5,13 +5,255 @@
 created on June 23, 2017
 """
 
-__all__ = ['DiskIntegratedModelClass', 'HG', 'HG12', 'HG1G2',
-           'DiskFunctionModel', 'LommelSeeliger', 'Lambert',
+__all__ = ['ref2mag', 'mag2ref', 'spline',
+           'DiskIntegratedModelClass', 'HG', 'HG12', 'HG1G2',
+           'DiskFunctionModel', 'LommelSeeliger', 'Lambert', 'LunarLambert',
            'PhaseFunctionModel', 'ROLOPhase',
            'ResolvedPhotometricModelClass', 'ROLO']
 
+import numpy as np
+from scipy.integrate import quad
+from astropy.modeling import FittableModel, Fittable1DModel, Fittable2DModel, Parameter
+from astropy import units
+
+
+def ref2mag(ref, radius, M_sun=None):
+    """Convert average bidirectional reflectance to reduced magnitude
+
+    Parameters
+    ----------
+    ref : num, astropy.units.Quantity
+        Average bidirectional reflectance
+    radius : num, astropy.units.Quantity
+        Radius of object
+    M_sun : number, optional
+        The magnitude of the Sun, default is -26.74
+
+    Returns
+    -------
+    The same type as input, reduced magnitude (at r=delta=1 au)
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> mag = ref2mag(0.1, 460)
+    >>> print('{0:.4}'.format(mag))
+    2.078
+    >>> mag = ref2mag(0.1, 460*u.km)
+    >>> print('{0:.4}'.format(mag))
+    2.078 mag
+    """
+
+    if M_sun is None:
+        M_sun = -26.74
+    Q = False
+    if isinstance(ref, units.Quantity):
+        ref = ref.value
+        Q = True
+    if isinstance(radius, units.Quantity):
+        radius = radius.to('km').value
+        Q = True
+    if isinstance(M_sun, units.Quantity):
+        M_sun = M_sun.to('mag').value
+        Q = True
+
+    mag = M_sun-2.5*np.log10(ref*np.pi*radius*radius*units.km.to('au')**2)
+    if Q:
+        return mag*units.mag
+    else:
+        return mag
+
+
+def mag2ref(mag, radius, M_sun=None):
+    """Convert reduced magnitude to average bidirectional reflectance
+
+    Parameters
+    ----------
+    mag : num, astropy.units.Quantity
+        Reduced magnitude
+    radius : num, astropy.units.Quantity
+        Radius of object
+    M_sun : number, optional
+        The magnitude of the Sun, default is -26.74
+
+    Returns
+    -------
+    The same type as input, average bidirectional reflectance
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> ref = mag2ref(2.08, 460)
+    >>> print('{0:.4}'.format(ref))
+    0.09981
+    >>> ref = mag2ref(2.08, 460*u.km)
+    >>> print('{0:.4}'.format(ref))
+    0.09981 1 / sr
+    """
+
+    if M_sun is None:
```

Comment:
I think that is better.  Fewer lines of code to test.

Reply:
The reason is because I don't want to explicitly set the value of `M_sun` in `DiskIntegratedModelClass.__init__()` or any other code that calls `ref2mag` or `mag2ref`, in order to avoid typos in the number and updating too much code if in the future the value of `M_sun` needs to be updated.